pkg/scheduler/core/generic_scheduler_test.go

```diff
@@ -814,21 +827,25 @@ func TestFindFitAllError(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	_, nodeToStatusMap, err := scheduler.findNodesThatFitPod(context.Background(), fwk, framework.NewCycleState(), &v1.Pod{})
+	_, diagnosis, err := scheduler.findNodesThatFitPod(context.Background(), fwk, framework.NewCycleState(), &v1.Pod{})
 
 	if err != nil {
 		t.Errorf("unexpected error: %v", err)
 	}
 
-	if len(nodeToStatusMap) != len(nodes) {
-		t.Errorf("unexpected failed status map: %v", nodeToStatusMap)
+	if len(diagnosis.NodeToStatusMap) != len(nodes) {
```

Comment:
can we replace all these lines with a single `cmp.Diff` for diagnosis?

Or is this an instance of #94695?

Reply:
I can use `cmd.Diff`. Will update.

Reply:
Ah hold on, unexported fields are not comparable, I will leave it as is and revisit once a general compared for `Status` in ready.

Reply:
That's what I thought. Leaving it for now is fine.

Reply:
Added a TODO at https://github.com/kubernetes/kubernetes/issues/94696#issuecomment-765701436.

Reply:
leave as comment in code

Reply:
Done.